website/docs/d/container_registry.html.md

```diff
@@ -0,0 +1,56 @@
+---
+layout: "digitalocean"
+page_title: "DigitalOcean: digitalocean_container_registry"
+sidebar_current: "docs-do-datasource-container-registry"
+description: |-
+  Get information on a container registry.
+---
+
+# digitalocean_container_registry
+
+Get information on a container registry. This data source provides the name as 
+configured on your DigitalOcean account. This is useful if the container 
+registry name in question is not managed by Terraform or you need validate if 
+the container registry exists in the account.
+
+An error is triggered if the provided container registry name does not exist.
+
+## Example Usage
+
+Get the container registry:
+
+```hcl
+data "digitalocean_container_registry" "example" {
+  name = "example"
+}
+```
+
+Use the endpoint and docker_credentials in the docker provider
+
+```
+provider "docker" {
+  host = "tcp:localhost:2376"
+
+  registry_auth {
+    address = data.digitalocean_container_registry.example.server_url
+    config_file_content = data.digitalocean_container_registry.example.docker_credentials
+  }
+}
+```
+
```

Comment:
I think it would be useful to also provide a full example for the K8s use case. Something like:
(The suggestion feature doesn't seem to like markdown inside of markdown...)

### Kubernetes Example

Combined with the Kubernetes Provider's [`kubernetes_secret` resource](https://www.terraform.io/docs/providers/kubernetes/r/secret.html), you can access the registry from inside your cluster:

```
data "digitalocean_container_registry" "example" {
  name = "example"
}

resource "digitalocean_kubernetes_cluster" "example" {
  name    = "example"
  region  = "nyc1"
  # Grab the latest version slug from `doctl kubernetes options versions`
  version = "1.16.6-do.0"

  node_pool {
    name       = "worker-pool"
    size       = "s-2vcpu-2gb"
    node_count = 3
  }
}

provider "kubernetes" {
  load_config_file = false
  host  = digitalocean_kubernetes_cluster.example.endpoint
  token = digitalocean_kubernetes_cluster.example.kube_config[0].token
  cluster_ca_certificate = base64decode(
    digitalocean_kubernetes_cluster.example.kube_config[0].cluster_ca_certificate
  )
}

resource "kubernetes_secret" "example" {
  metadata {
    name = "docker-cfg"
  }

  data = {
    ".dockerconfigjson" = data.digitalocean_container_registry.example.docker_credentials
  }

  type = "kubernetes.io/dockerconfigjson"
}
```

Reply:
Done and committed.